From 73c196e587389f52e4f01979647551384135ea4b Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Thu, 9 Oct 2014 21:00:11 -0400
Subject: [fs] xfs: fix wrong errno from xfs_initxattrs

Message-id: <1412888441-4181-18-git-send-email-sandeen@redhat.com>
Patchwork-id: 97322
O-Subject: [RHEL7.1 PATCH 17/47] [fs] xfs: fix wrong errno from xfs_initxattrs
Bugzilla: 1145837
RH-Acked-by: Brian Foster <bfoster@redhat.com>
RH-Acked-by: Dave Chinner <dchinner@redhat.com>

Bugzilla: 1145837
Upstream Status: Committed upstream
Build Info: https://brewweb.devel.redhat.com/taskinfo?taskID=8082426
Tested: locally using xfstests

Thanks,
-Eric

    commit a5a14de22e8afd771775c7106b3b081c23bac783
    Author: Dave Chinner <dchinner@redhat.com>
    Date:   Thu May 15 09:22:21 2014 +1000

    xfs: fix wrong errno from xfs_initxattrs

    Signed-off-by: Dave Chinner <dchinner@redhat.com>
    Reviewed-by: Jie Liu <jeff.liu@oracle.com>
    Signed-off-by: Dave Chinner <david@fromorbit.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/xfs/xfs_iops.c b/fs/xfs/xfs_iops.c
index 2662ea1..a713fca 100644
--- a/fs/xfs/xfs_iops.c
+++ b/fs/xfs/xfs_iops.c
@@ -59,8 +59,8 @@ xfs_initxattrs(
  int   error = 0;
 
  for (xattr = xattr_array; xattr->name != NULL; xattr++) {
-  error = xfs_attr_set(ip, xattr->name, xattr->value,
-         xattr->value_len, ATTR_SECURE);
+  error = -xfs_attr_set(ip, xattr->name, xattr->value,
+          xattr->value_len, ATTR_SECURE);
   if (error < 0)
    break;
  }
@@ -80,8 +80,8 @@ xfs_init_security(
  struct inode *dir,
  const struct qstr *qstr)
 {
- return security_inode_init_security(inode, dir, qstr,
-         &xfs_initxattrs, NULL);
+ return -security_inode_init_security(inode, dir, qstr,
+          &xfs_initxattrs, NULL);
 }
 
 static void
-- 
1.7.1